File: app/src/main/java/in/woloo/www/search/SearchWolooActivity.java
package in.woloo.www.search;

import static in.woloo.www.utils.AppConstants.NEARBY_WOLOO_OFFERS;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;
import android.widget.FrameLayout;

import butterknife.BindView;
import butterknife.ButterKnife;
import in.woloo.www.R;
import in.woloo.www.common.CommonUtils;
import in.woloo.www.scan_qr_code.QRcodeScannerActivity;
import in.woloo.www.search.fragments.WolooSearchFragment;
import in.woloo.www.utils.Logger;

public class SearchWolooActivity extends AppCompatActivity {

    public static final String ARG_SHOW_OFFERS = "ARG_SHOW_OFFERS";
    @BindView(R.id.frmFragment)
    FrameLayout frmFragment;
    public static String TAG = SearchWolooActivity.class.getSimpleName();

    /*calling on onCreate*/
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_search_woloo);
        ButterKnife.bind(this);
        initViews();
        Logger.i(TAG, "onCreate");
    }

    /*calling on initViews*/
    private void initViews() {
        Logger.i(TAG, "initViews");
        try {
            if (getIntent() != null && getIntent().getBooleanExtra(ARG_SHOW_OFFERS, false)) {
                loadFragment(WolooSearchFragment.newInstance(true));
            } else {
                loadFragment(WolooSearchFragment.newInstance(false));
            }
        } catch (Exception ex) {
             CommonUtils.printStackTrace(ex);
        }
    }

    /*calling on loadFragment*/
    public void loadFragment(Fragment fragment) {
        try {
            Logger.i(TAG, "loadFragment");
            FragmentManager fragmentManager = ((AppCompatActivity) this).getSupportFragmentManager();
            FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
            fragmentTransaction.replace(R.id.frmFragment, fragment);
            fragmentTransaction.commit();
        } catch (Exception ex) {
             CommonUtils.printStackTrace(ex);
        }
    }
}

File: app/src/main/java/in/woloo/www/application_kotlin/presentation/activities/loo_discovery/SearchWolooActivity.kt
package `in`.woloo.www.application_kotlin.presentation.activities.loo_discovery

import android.os.Bundle
import android.widget.FrameLayout
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import butterknife.BindView
import butterknife.ButterKnife
import `in`.woloo.www.R
import `in`.woloo.www.common.CommonUtils
import `in`.woloo.www.application_kotlin.presentation.fragments.WolooSearchFragment
import `in`.woloo.www.utils.Logger

class SearchWolooActivity : AppCompatActivity() {
    @JvmField
    @BindView(R.id.frmFragment)
    var frmFragment: FrameLayout? = null

    /*calling on onCreate*/
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_search_woloo)
        ButterKnife.bind(this)
        initViews()
        Logger.i(TAG, "onCreate")
    }

    /*calling on initViews*/
    private fun initViews() {
        Logger.i(TAG, "initViews")
        try {
            if (intent != null && intent.getBooleanExtra(ARG_SHOW_OFFERS, false)) {
                loadFragment(WolooSearchFragment.newInstance(true))
            } else {
                loadFragment(WolooSearchFragment.newInstance(false))
            }
        } catch (ex: Exception) {
            CommonUtils.printStackTrace(ex)
        }
    }

    /*calling on loadFragment*/
    fun loadFragment(fragment: Fragment?) {
        try {
            Logger.i(TAG, "loadFragment")
            val fragmentManager = (this as AppCompatActivity).supportFragmentManager
            val fragmentTransaction = fragmentManager.beginTransaction()
            fragmentTransaction.replace(R.id.frmFragment, fragment!!)
            fragmentTransaction.commit()
        } catch (ex: Exception) {
            CommonUtils.printStackTrace(ex)
        }
    }

    companion object {
        const val ARG_SHOW_OFFERS = "ARG_SHOW_OFFERS"
        var TAG = SearchWolooActivity::class.java.simpleName
    }
}